app/src/main/java/org/mozilla/tv/firefox/ScreenControllerStateMachine.kt

```diff
@@ -0,0 +1,88 @@
+
+// TODO add this stuff
+
+package org.mozilla.tv.firefox
+
+import android.support.annotation.VisibleForTesting
+
+/**
+ * TODO
+ */
+class ScreenControllerStateMachine(private val currentUrlIsHome: () -> Boolean) {
+
+    enum class ActiveScreen {
+        NAVIGATION_OVERLAY, WEB_RENDER, POCKET, SETTINGS
+    }
+
+    enum class Transition {
+        ADD_OVERLAY, REMOVE_OVERLAY, ADD_POCKET, REMOVE_POCKET, ADD_SETTINGS, REMOVE_SETTINGS, EXIT_APP, NO_OP
+    }
+
+    var currentActiveScreen: ActiveScreen = ActiveScreen.NAVIGATION_OVERLAY
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        set
+
+    fun menuPress(): Transition {
+        return when(currentActiveScreen) {
+            ActiveScreen.NAVIGATION_OVERLAY -> {
+                return if (currentUrlIsHome()) {
+                    Transition.NO_OP
```

Comment:
What's the `NO_OP` state?

Reply:
The action here is that nothing happens, so we made it a NO_OP - but maybe that's not a very clear state name hm.